src/mod_stream_management.erl

```diff
@@ -119,6 +157,34 @@ set_resume_timeout(ResumeTimeout) ->
 %% API for `ejabberd_c2s'
 %%
 
+-spec make_smid() -> smid().
+make_smid() ->
+    base64:encode(crypto:strong_rand_bytes(21)).
+
+%% Getters
+-spec get_session_from_smid(SMID :: smid()) ->
+    ejabberd_sm:sid() | non_neg_integer() | smid_not_found.
```

Comment:
The return value is quite confusing. As far as I understand we need sth like this:
`{ok, ejabberd_sm:sid()} | {smid_not_found, maybe_stale_h()}`

Reply:
Or something like: `#{sid => Sid} | #{stale_h => Stale} | #{}`